code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm

```diff
@@ -260,56 +255,58 @@
 	description = "Makes you nearly impervious to stuns and grants a stamina regeneration buff, but you will be a nearly uncontrollable tramp-bearded raving lunatic."
 	reagent_state = LIQUID
 	color = "#60A584" // rgb: 96, 165, 132
+	metabolization_rate = 0.5 * REAGENTS_METABOLISM
 	overdose_threshold = 15
 	addiction_threshold = 10
+	stun_threshold = 2
+	speedboost = VERY_FAST + IGNORE_SLOWDOWN
 
 
 /datum/reagent/drug/bath_salts/on_mob_life(mob/living/M)
 	var/high_message = pick("You feel your grip on reality loosening.", "You feel like your heart is beating out of control.", "You feel as if you're about to die.")
 	if(prob(15))
 		M << "<span class='notice'>[high_message]</span>"
-	if(current_cycle >= 5)
+	if(!(M.stunned || M.weakened || M.paralysis))
+		stun_timer++
+		metabolization_rate = 0.5* REAGENTS_METABOLISM
+		M.adjustStaminaLoss(-4)
+	else
+		metabolization_rate = REAGENTS_METABOLISM
```

Comment:
Could you not use initial(metabolization_rate) is your multiplier?
So if someone wants to change the metabolizationrate, they only need to change the value at the definitions of the reagent?